instance/models/mixins/openedx_database.py

```diff
@@ -228,6 +228,13 @@ def set_field_defaults(self):
         But since we add suffixes to mysql_user to generate unique user names
         for different services (e.g. xqueue) we don't want to use the maximum length here.
         """
+        # FIXME: Remove first check to make sure any instance can switch to using persistent databases?
```

Comment:
@pomegranited 

> It looks like this this code already does allow us to switch from ephemeral to persistent,  by just setting `use_ephemeral_databases = False`

Yes it does, but `set_field_defaults` is currently only called if an instance has not been saved to the database, see `ValidateModelMixin` > `clean_fields` > [L94](https://github.com/open-craft/opencraft/blob/master/instance/models/utils.py#L94). This means that if a field doesn't get set via one of the `set_field_defaults` methods, we have to set it manually afterwards. And that's the reason why the code sets `self.mysql_user`, `self.mongo_user`, etc. below without checking the value of `self.use_ephemeral_databases` first -- an instance that uses ephemeral databases doesn't use/need these credentials, but we want them to be available right away in case we decide to "flip the switch" (set `use_ephemeral_databases` to `False`) on the instance.

I think it makes sense to have the same behavior for MySQL and MongoDB servers, so I'm going to remove the first check.